src/Server.js

```diff
@@ -54,7 +54,25 @@ function middleware(bundler) {
   };
 }
 
-function serve(bundler, port) {
+function checkPortAvailabilty(port) {
+  return new Promise((resolve, reject) => {
+    let server = http.createServer().listen(port);
+    server.once('error', err => {
+      resolve(false);
+    });
+    server.once('listening', connection => {
+      server.close(() => {
+        resolve(true);
+      });
+    });
+  });
+}
```

Comment:
Do we need to create a whole separate server here? Could we use the normal server made below to do this, and handle errors?

Reply:
IMO, moving the logic to check for a free port to its own method seems apt, although that's just a matter of preference. 😅

Reply:
If we're not opposed to _another_ dependency, something like [get-port](https://www.npmjs.com/package/get-port) would simplify things. It lets you pass in the default, and returns an ephemeral port if it isn't available. 

```js
const port = getPort({port: 1234});
```

Reply:
@brandon93s code wise this package does exactly the same so it should not really make a difference and would clean up the code a bit i'll pivot towards it, keeps things clean

Reply:
fixed in aaa5308c2dac6ddb3b637d0c9bbaff8f12ebca97